File: app/src/main/java/com/example/purepawapp/ui/onboarding/OnboardingPagerAdapter.java
package com.example.purepawapp.ui.onboarding;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.purepawapp.databinding.ItemOnboardingPageBinding;

import java.util.List;

public class OnboardingPagerAdapter extends RecyclerView.Adapter<OnboardingPagerAdapter.PageViewHolder> {

    private final List<OnboardingPage> pages;

    public OnboardingPagerAdapter(List<OnboardingPage> pages) {
        this.pages = pages;
    }

    public static class PageViewHolder extends RecyclerView.ViewHolder {
        final ItemOnboardingPageBinding binding;

        PageViewHolder(ItemOnboardingPageBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    @NonNull
    @Override
    public PageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemOnboardingPageBinding binding = ItemOnboardingPageBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new PageViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull PageViewHolder holder, int position) {
        OnboardingPage page = pages.get(position);
        holder.binding.tvPageTitle.setText(page.getTitle());
        holder.binding.tvPageDescription.setText(page.getDescription());
    }

    @Override
    public int getItemCount() {
        return pages.size();
    }
}

File: app/src/main/java/com/example/purepawapp/ui/onboarding/OnboardingPagerAdapter.kt
package com.example.purepawapp.ui.onboarding

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.purepawapp.databinding.ItemOnboardingPageBinding

data class OnboardingPage(val title: String, val description: String)

class OnboardingPagerAdapter(private val pages: List<OnboardingPage>) :
    RecyclerView.Adapter<OnboardingPagerAdapter.PageViewHolder>() {

    inner class PageViewHolder(val binding: ItemOnboardingPageBinding) :
        RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PageViewHolder {
        val binding = ItemOnboardingPageBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return PageViewHolder(binding)
    }

    override fun onBindViewHolder(holder: PageViewHolder, position: Int) {
        val page = pages[position]
        holder.binding.tvPageTitle.text = page.title
        holder.binding.tvPageDescription.text = page.description
    }

    override fun getItemCount(): Int = pages.size
}
